viewer/packages/cad-model/src/picking/PickingHandler.ts

```diff
@@ -92,16 +92,26 @@ export class PickingHandler {
   ): Promise<IntersectCadNodesResult[]> {
     const results: IntersectCadNodesResult[] = [];
     const release = await this._mutex.acquire();
+    let cadNodesVisibility: boolean[];
 
     try {
       for (const cadNode of cadNodes) {
+        // Make current CAD node visible & hide others
+        cadNodesVisibility = cadNodes.map(p => {
+          const previousVisibility = p.visible;
+          p.visible = false;
+          return previousVisibility;
+        });
```

Comment:
Why do this for every model? Should only be done once outside the for loop. This will not work as you expect, since only the state after the last model is processed is used.

Reply:
Should be easily unit testable.